aea/helpers/profiling.py

```diff
@@ -86,56 +88,50 @@ class Profiling(Runnable):
 
     def __init__(
         self,
+        types_to_track: List[Type],
         period: int = 0,
-        objects_instances_to_count: List[Type] = None,
-        objects_created_to_count: List[Type] = None,
         output_function: Callable[[str], None] = lambda x: print(x, flush=True),
     ) -> None:
         """
         Init profiler.
 
         :param period: delay between profiling output in seconds.
-        :param objects_instances_to_count: object to count
-        :param objects_created_to_count: object created to count
+        :param types_to_track: object types to count
         :param output_function: function to display output, one str argument.
         """
         if period < 1:  # pragma: nocover
-            raise ValueError("Period should be at least 1 second!")
+            raise ValueError("Profiling frequency should be at least 1 second!")
         super().__init__(threaded=True)
         self._period = period
         self._start_ts = time.time()
-        self._objects_instances_to_count = objects_instances_to_count or []
-        self._objects_created_to_count = objects_created_to_count or []
+        self._types_to_track = types_to_track
         self._output_function = output_function
-        self._counter: Dict[Type, int] = Counter()
+        self.object_counts: Dict[Type, List[int]] = dict(
+            zip(types_to_track, [[0, 0]] * len(types_to_track))
```

Comment:
you are creating aliases here, a particularly nasty little thing in python sometimes. yes, multiplying a regular list works just fine if you want to extend it to a desired length. It does not, however, work the same way for nested lists. More specifically, it only works when the list (or tuple) contains primitive types.

to illustrate with an example:
```
d = dict(zip("abc", [[0, 0]] * 3))
d['a'][0] += 1
print(d)
>>> {'a': [1, 0], 'b': [1, 0], 'c': [1, 0]}
```

suggested: 
```
d = {x : [0, 0] for x in "abc"}
d['a'][0] += 1
print(d)
>>> {'a': [1, 0], 'b': [0, 0], 'c': [0, 0]}
```